vlog4j-core/src/test/java/org/semanticweb/vlog4j/core/model/RuleImplTest.java

```diff
@@ -137,4 +138,21 @@ public void noUnsafeVariables() {
 		Expressions.makeRule(literal1, literal2);
 	}
 
+	@Test
+	public void testtoString() {
+		final Variable x = Expressions.makeUniversalVariable("X");
+		final Variable y = Expressions.makeExistentialVariable("Y");
+		final Variable z = Expressions.makeUniversalVariable("Z");
+		final Constant c = Expressions.makeAbstractConstant("c");
+		final PositiveLiteral atom1 = Expressions.makePositiveLiteral("p", x, c);
```

Comment:
I would include different types of Literals and Constants in this test.

Reply:
I guess This is not mine as well, I added tests for different type of constants.

Reply:
the rule calls the tostring of every literal and constant type it has, so if you have a test for all the types anywhere else that works, be sure that it will appear the same anywhere else.